src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -2782,6 +2782,176 @@ export default class EthereumApi implements Api {
       overrides
     );
   }
+
+  /**
+   * Retrieves past blocks
+   *
+   * @param blockCount - Requested range of blocks. Will return less than the requested range if not all blocks are available.
+   * @param newestBlock - Highest block of the requested range.
+   * @param rewardPercentiles - (Optional) A monotonically increasing list of percentile values. For each block in the requested range, the transactions will be sorted in ascending order by effective tip per gas and the coresponding effective tip for the percentile will be determined, accounting for gas consumed.
+   *
+   * @returns The fee history for the returned block range. This can be a subsection of the requested range if not all blocks are available. Includes the oldestBlock, baseFeePerGas, gasUsedRatio, and percentile rewards.
+   * @example for: blockCount: 2, newestBlock: 0x10, rewardPercentiles: [10, 25, 50]
+   * {
+   *   oldestBlock: "0x10",
+   *   reward:[["0x59682f00","0x2a7b749c11","0x2a7b749c11"]],
+   *   baseFeePerGas:["0x40c9459e7","0x420068cc1","0x420068cc1"]
+   *   gasUsedRatio: [0.5750394293729167, 0.50]
+   * };
+   *
+   */
+  @assertArgLength(2, 3)
+  async eth_feeHistory(
+    blockCount: QUANTITY,
+    newestBlock: QUANTITY | Ethereum.Tag,
+    rewardPercentiles?: number[]
+  ): Promise<Ethereum.FeeHistoryResult> {
+    let reward = [];
+    const baseFeePerGas = [];
+    const gasUsedRatio = [];
+    const blockchain = this.#blockchain;
+    const PAD_PRECISION = 16; // Matching geth/infura decimal precision formatting
+    const PRECISION = 10 ** PAD_PRECISION;
+    const PRECISION_BIG_INT = Quantity.from(PRECISION).toBigInt();
```

Comment:
I'd avoid using `Quantity` here altogether - although it's tempting, because it's super convenient, it's intended for the very specific use case of RPC Quantity data types (there's some special cases and rules specific to this).

When you do `Quantity.from(xxx)` it validates and converts `xxx` to a `Buffer` as it's internal representation, which `toBigInt()` then converts the buffer value to `BigInt`.

These conversions are fairly expensive, and `BigInt` constructors themselves are relatively expensive (~30x slower than bigint constants) although probably not really enough for us to care too much.

I'd encourage using `BigInt` constants as much as possible (rather than `Number` constants and `BigInt` constructor).

Reply:
This is perfect. I wasn't sure if I should use `Quantity` like a swiss army knife and this def helped improve what was in place.